SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp

```diff
@@ -193,17 +171,90 @@ bool GenerateLaunchConfig(const FString& LaunchConfigPath, const FSpatialLaunchC
 			return false;
 		}
 
+		if (bGenerateCloudConfig)
+		{
+			return ConvertToClassicConfig(LaunchConfigPath, InLaunchConfigDescription);
+		}
+
 		return true;
 	}
 
 	return false;
 }
 
-bool ValidateGeneratedLaunchConfig(const FSpatialLaunchConfigDescription& LaunchConfigDesc, const FWorkerTypeLaunchSection& InWorker)
+bool ConvertToClassicConfig(const FString& LaunchConfigPath, const FSpatialLaunchConfigDescription* InLaunchConfigDescription)
 {
-	const USpatialGDKSettings* SpatialGDKRuntimeSettings = GetDefault<USpatialGDKSettings>();
+	// The new runtime binary handles the config conversion. We just need to pass it the parameters.
+	// All `runtime_flags` are converted to `legacy_flags` in the conversion process.
+	// `max_concurrent_workers` is also converted to a legacy flag.
+	// The output config file is called `launch_config.json` when printing and we must provide a path to a folder for it to be saved to.
+	// The output config file will replace the generated standalone config file.
+	// We do not export the worker configurations as we already generate these for the user still.
+
+	FString LaunchConfigDir = FPaths::GetPath(LaunchConfigPath);
 
-	if (const FString* EnableChunkInterest = LaunchConfigDesc.World.LegacyFlags.Find(TEXT("enable_chunk_interest")))
+	// runtime.exe --export-classic-config=classic_config_dir --config=Game\Intermediate\Improbable\Control_Small_LocalLaunchConfig.json
+	// --launch-template=w2_r0500_e5 --export-worker-configuration=true
+	FString ConversionArgs =
+		FString::Printf(TEXT("--export-classic-config=\"%s\" --config=\"%s\" --launch-template=%s --export-worker-configuration=false"),
+						*LaunchConfigDir, *LaunchConfigPath, *InLaunchConfigDescription->GetTemplate());
+
+	FString Output;
+	int32 ExitCode;
+
+	const USpatialGDKEditorSettings* SpatialGDKEditor = GetDefault<USpatialGDKEditorSettings>();
+	FString RuntimePath =
+		SpatialGDKServicesConstants::GetRuntimeExecutablePath(SpatialGDKEditor->GetSelectedRuntimeVariantVersion().GetVersionForLocal());
+
+	FSpatialGDKServicesModule::ExecuteAndReadOutput(RuntimePath, ConversionArgs, LaunchConfigDir, Output, ExitCode);
+
+	if (!ExitCode == SpatialGDKServicesConstants::ExitCodeSuccess)
```

Comment:
```suggestion
	if (ExitCode != SpatialGDKServicesConstants::ExitCodeSuccess)
```

Reply:
I have no idea how I wrote this, but I love it.

Reply:
Magically I think !ExitCode does actually work, just seems weird

Reply:
It's magical :D